libpromises/evalfunction.c

```diff
@@ -2775,6 +2847,22 @@ static FnCallResult FnCallMapData(EvalContext *ctx, ARG_UNUSED const Policy *pol
         return FnFailure();
     }
 
+    if (mapdatamode && jqmode)
+    {
+        // TODO: make this customizable
```

Comment:
Another proposal I made in the ticket https://dev.cfengine.com/issues/7435 was to support the syntax `mapdata("|/usr/bin/jq", ...)` which would make this moot. The user decides what to pipe.